kernel/kernel-api/src/main/java/io/delta/kernel/internal/table/ResolvedTableBuilderImpl.java

```diff
@@ -106,7 +111,41 @@ public ResolvedTableInternal build(Engine engine) {
 
   private void validateInputOnBuild() {
     checkArgument(ctx.versionOpt.orElse(0L) >= 0, "version must be >= 0");
-    // TODO: logData only ratified staged commits
-    // TODO: logData sorted and contiguous
+    validateProtocolAndMetadataOnlyIfVersionProvided();
+    validateProtocolRead();
+    validateLogDatasOnlyRatifiedCommits();
```

Comment:
Can you add a TODO to support other type? Maybe only RatifiedCommits are supported in the short term but this validation now give me a wrong sense that resolved table is only working with the specific type.